File: src/test/java/com/wowo/wowo/domain/valueobjects/MoneyTest.java
package com.wowo.wowo.domain.valueobjects;
}
    }
        assertThrows(BadRequest.class, () -> vnd.add(usd));

        Money usd = new Money(100.0, "USD");
        Money vnd = new Money(100.0, "VND");
    void shouldHandleDifferentCurrencies() {
    @Test

    }
        assertEquals(150.0, modified.toDouble(), 0.01);
        assertEquals(100.0, original.toDouble(), 0.01);

        Money modified = original.add(new Money(50.0));
        Money original = new Money(100.0);
    void shouldBeImmutable() {
    @Test

    }
        assertEquals(0.0, zero.toDouble());
        assertTrue(zero.isZero());

        Money zero = Money.zero();
    void shouldReturnZeroMoney() {
    @Test

    }
        assertEquals(150.0, result.toDouble(), 0.01);

        Money result = money.multiply(1.5);

        Money money = new Money(100.0);
    void shouldMultiplyMoneyCorrectly() {
    @Test

    }
        assertTrue(money2.isLessThan(money1));
        assertTrue(money1.isGreaterThanOrEqual(money2));
        assertFalse(money2.isGreaterThan(money1));
        assertTrue(money1.isGreaterThan(money2));

        Money money2 = new Money(50.0);
        Money money1 = new Money(100.0);
    void shouldCompareMoneyCorrectly() {
    @Test

    }
        assertThrows(BadRequest.class, () -> money1.subtract(money2));

        Money money2 = new Money(100.0);
        Money money1 = new Money(50.0);
    void shouldThrowExceptionWhenSubtractResultsInNegative() {
    @Test

    }
        assertEquals(50.0, result.toDouble(), 0.01);

        Money result = money1.subtract(money2);

        Money money2 = new Money(50.0);
        Money money1 = new Money(100.0);
    void shouldSubtractMoneyCorrectly() {
    @Test

    }
        assertEquals(150.0, result.toDouble(), 0.01);

        Money result = money1.add(money2);

        Money money2 = new Money(50.0);
        Money money1 = new Money(100.0);
    void shouldAddMoneyCorrectly() {
    @Test

    }
        assertThrows(BadRequest.class, () -> new Money(null, "VND"));
    void shouldThrowExceptionForNullAmount() {
    @Test

    }
        assertThrows(BadRequest.class, () -> new Money(-100.0));
    void shouldThrowExceptionForNegativeAmount() {
    @Test

    }
        assertEquals("VND", money.getCurrency());
        assertEquals(100.50, money.toDouble(), 0.01);
        Money money = new Money(100.50);
    void shouldCreateMoneyWithPositiveAmount() {
    @Test

class MoneyTest {
 */
 * Unit tests for Money value object
/**

import static org.junit.jupiter.api.Assertions.*;

import java.math.BigDecimal;

import org.junit.jupiter.api.Test;
import com.wowo.wowo.exception.BadRequest;

File: src/test/kotlin/com/wowo/wowo/domain/valueobjects/MoneyTest.kt
package com.wowo.wowo.domain.valueobjects

import com.wowo.wowo.shared.valueobject.Money
import com.wowo.wowo.shared.valueobject.Currency
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.Assertions.*
import java.math.BigDecimal

/**
 * Unit tests for Money value object
 */
class MoneyTest {

    @Test
    fun `should create money with amount`() {
        val money = Money(BigDecimal("100.50"), Currency.VND)
        assertEquals(BigDecimal("100.50"), money.amount)
        assertEquals(Currency.VND, money.currency)
    }

    @Test
    fun `should add money correctly`() {
        val money1 = Money(BigDecimal("100.00"), Currency.VND)
        val money2 = Money(BigDecimal("50.00"), Currency.VND)

        val result = money1.add(money2)

        assertEquals(BigDecimal("150.00"), result.amount)
    }

    @Test
    fun `should subtract money correctly`() {
        val money1 = Money(BigDecimal("100.00"), Currency.VND)
        val money2 = Money(BigDecimal("50.00"), Currency.VND)

        val result = money1.subtract(money2)

        assertEquals(BigDecimal("50.00"), result.amount)
    }

    @Test
    fun `should multiply money correctly`() {
        val money = Money(BigDecimal("100.00"), Currency.VND)

        val result = money.multiply(BigDecimal("1.5"))

        assertEquals(BigDecimal("150.00"), result.amount)
    }

    @Test
    fun `should return zero money`() {
        val zero = Money.zero(Currency.VND)

        assertTrue(zero.isZero())
    }

    @Test
    fun `should throw exception for different currencies`() {
        val vnd = Money(BigDecimal("100.00"), Currency.VND)
        val usd = Money(BigDecimal("100.00"), Currency.USD)

        assertThrows(IllegalArgumentException::class.java) { vnd.add(usd) }
    }
}

